overlord/snapstate/handlers.go

```diff
@@ -383,7 +383,17 @@ func (m *SnapManager) undoPrepareSnap(t *state.Task, _ *tomb.Tomb) error {
 		extra["UbuntuCoreTransitionCount"] = strconv.Itoa(ubuntuCoreTransitionCount)
 	}
 
-	if !settings.ProblemReportsDisabled(st) {
+	// Only report and error if there is an actual error in the change,
+	// we could undo things because the user canceled the change.
+	isErr := func() bool {
```

Comment:
Why is this an inline function?  If this is just the equivalent of the following, I'd keep it simple:
```
var isErr bool
for _, tt := range t.Change.Tasks() {
    if tt.Status() == state.ErrorStatus {
        isErr = true
        break
    }
}